Thanks for the PR.

It seems that the function is missing in some version (?) of SDL2

```
/usr/bin/clang++-9 -DFSUAE -I/home/runner/work/quaesar/quaesar/uae_src/include -I/home/runner/work/quaesar/quaesar/uae_src -I/home/runner/work/quaesar/quaesar/src -I/home/runner/work/quaesar/quaesar/. -I/usr/include/SDL2 -I/home/runner/work/quaesar/quaesar/external/ADFlib/src -I/home/runner/work/quaesar/quaesar/temp/external/zlib -I/home/runner/work/quaesar/quaesar/external/zlib -std=gnu++17 -DUAE=1 -D_cdecl= -DFILEFLAG_WRITE=1 -DOS_NAME=\"linux\" -DUSHORT=uint16_t -DHWND=uint32_t -DHRESULT=uint32_t -DWPARAM=uint16_t -DLPARAM=uint32_t -DFILEFLAG_DIR=1 -D_ftelli64=ftello64 -D_fseeki64=fseeko64 -MD -MT CMakeFiles/quaesar.dir/src/sounddep/sound.cpp.o -MF CMakeFiles/quaesar.dir/src/sounddep/sound.cpp.o.d -o CMakeFiles/quaesar.dir/src/sounddep/sound.cpp.o -c /home/runner/work/quaesar/quaesar/src/sounddep/sound.cpp
/home/runner/work/quaesar/quaesar/src/sounddep/sound.cpp:370:9: error: use of undeclared identifier 'SDL_GetDefaultAudioInfo'
    if (SDL_GetDefaultAudioInfo(&cur_driver, &spec, 0) == 0) {
```

Any idea here?

--- a/src/sounddep/sound.cpp
+++ b/src/sounddep/sound.cpp
@@ -365,15 +365,19 @@ void resume_sound_device(struct sound_data* sd) {
 
 int get_default_audio_device() {
     int device_idx = -1;
+#if SDL_VERSION_ATLEAST(2, 24, 0)
     SDL_AudioSpec spec;
-    char* cur_driver = nullptr;
-    if (SDL_GetDefaultAudioInfo(&cur_driver, &spec, 0) == 0) {
+    char* default_device_name = nullptr;
+    if (SDL_GetDefaultAudioInfo(&default_device_name, &spec, 0) == 0) {
         for (int i = 0; i < num_sound_devices; i++) {
-            if (strcmp(sound_devices[i]->name, cur_driver) == 0)
-                device_idx = i;
+            if (strcmp(sound_devices[i]->name, default_device_name) != 0)
+                continue;
+            device_idx = i;
+            break;
         }
-        SDL_free(cur_driver);
+        SDL_free(default_device_name);
     }
+#endif
     return device_idx;
 }
 

